bundles/automation/org.eclipse.smarthome.automation.importers/src/main/java/org/eclipse/smarthome/automation/internal/importers/AbstractImporter.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.internal.importers;
+
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.automation.RuleProvider;
+import org.eclipse.smarthome.automation.parser.Parser;
+import org.eclipse.smarthome.automation.template.TemplateProvider;
+import org.eclipse.smarthome.automation.type.ModuleTypeProvider;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is base for {@link RuleProvider}, {@link ModuleTypeProvider} and {@link TemplateProvider} which are
```

Comment:
Issue #1569 only speaks about module types and templates, but not about rules. Note that we have import mechanisms for rules already in place, which add them to the internal database. I would therefore reduce the scope here to only handle module types and templates.